### PR TITLE
Spk dev: display BASE_URL to the user.

### PR DESCRIPTION
**Pull request by zenhack**
Most of the diff (first commit) is just moving the config parsing into a separate file; the second patch just reads the value from the config and includes it in the message.

### CODE REVIEW

File: src/sandstorm/config.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  kj::Maybe<kj::String> stripePublicKey = nullptr;
+};
+
+// Read and return the config file from `path`.
+//
+// If `parseUids` is true, we initialize `uids` from SERVER_USER.  This requires shelling
+// out to id(1).  If false, we ignore SERVER_USER.
+Config readConfig(const char *path, bool parseUids);
+
+};
```
Comment (kentonv) [inline src/sandstorm/config.h]: Nitpick: Should be:

```
}  // namespace sandstorm
```

Comment (kentonv) [reply src/sandstorm/config.h]: Also in the `.c++` file.

File: src/sandstorm/run-bundle.c++ [hunk 2]
```diff
@@ -1689,128 +1511,12 @@ private:
   }
 
   Config readConfig() {
-    // Read and return the config file.
-    //
-    // If parseUids is true, we initialize `uids` from SERVER_USER.  This requires shelling
```
Comment (kentonv) [inline src/sandstorm/run-bundle.c++]: So the code had a comment mentioning `parseUids`, but no such option actually existed?

But you've added this option in the factored-out code?

Comment (zenhack) [reply src/sandstorm/run-bundle.c++]: Correct.

File: src/sandstorm/config.h (hunk 1 above)
Comment (zenhack) [reply src/sandstorm/config.h]: Done.

File: src/sandstorm/run-bundle.c++ (hunk 2 above)
Comment (kentonv) [reply src/sandstorm/run-bundle.c++]: Does the UID parsing not work in the `spk dev` use case, or did you only skip it because it's not needed?

**Comment by zenhack** [comment]
Quoting Kenton Varda (2020-03-15 20:19:08)

>    Does the UID parsing not work in the spk dev use case, or did you only
>    skip it because it's not needed?

The latter. I don't have a strong opinion about this though.

File: src/sandstorm/config.h (hunk 1 above)
Comment (kentonv) [reply src/sandstorm/config.h]: Looks like you left in the semicolons. FWIW the semicolon is redundant here -- though I'm not going to hold up merge on something like that.

**Comment by kentonv** [comment]
> The latter. I don't have a strong opinion about this though.

OK, I don't have a strong opinion either, was just curious.

The CI failure is my fault, fixed by #3270. Interesting that the CI build for pull requests merges with master, rather than test your exact commit.

**Pull request merged**